### PR TITLE
Feature/eicnet 2468 group invitations migration

### PR DESCRIPTION
**Pull request by lramarojaona**
### Improvements

- Added a `hook_mail_alter` to prevent emails to be sent out during migrations

### Tests

- [x] Run `upgrade_d7_group_pending_invitations`. You should get 1949/1952migrated items
- [x] Run `upgrade_d7_group_declined_invitations`. There is no items to migrate at this moment
- [x] Run `upgrade_d7_group_pending_visitor_invitations`. You should get 875/876 migrated items

### CODE REVIEW

**Review by 4hardfire — COMMENTED** [review]
- Migrate message for upgrade_d7_group_pending_invitations: `upgrade_d7_group_pending_invitations:uid: User not found`
- Migrate message for upgrade_d7_group_pending_visitor_invitations: `upgrade_d7_group_pending_visitor_invitations:gid: Group not found`. This is wrong because the group has been migrated (NID = 10010, Bundle = "event_site")

File: config/sync/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      method: row
+      message: 'User not found'
+  gid:
+    -
+      plugin: migration_lookup
+      source: gid
+      migration:
+        - upgrade_d7_node_complete_group
+        - upgrade_d7_node_complete_organisation
+        - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline config/sync/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml:48]: This should be `upgrade_d7_node_complete_event_site`

File: config/sync/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  created: created
+  changed: created
+destination:
+  plugin: 'entity:group_content'
+migration_dependencies:
+  required:
+    - upgrade_d7_user
+    - upgrade_d7_node_complete_group
+    - upgrade_d7_node_complete_organisation
+    - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline config/sync/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml:78]: This should be `upgrade_d7_node_complete_event_site`

File: config/sync/migrate_plus.migration.upgrade_d7_group_pending_visitor_invitations.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      method: row
+      message: 'User not found'
+  gid:
+    -
+      plugin: migration_lookup
+      source: inv_group_id
+      migration:
+        - upgrade_d7_node_complete_group
+        - upgrade_d7_node_complete_organisation
+        - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline config/sync/migrate_plus.migration.upgrade_d7_group_pending_visitor_invitations.yml:44]: This should be `upgrade_d7_node_complete_event_site`

File: config/sync/migrate_plus.migration.upgrade_d7_group_pending_visitor_invitations.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  created: inv_created
+  changed: inv_updated
+destination:
+  plugin: 'entity:group_content'
+migration_dependencies:
+  required:
+    - upgrade_d7_user
+    - upgrade_d7_node_complete_group
+    - upgrade_d7_node_complete_organisation
+    - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline config/sync/migrate_plus.migration.upgrade_d7_group_pending_visitor_invitations.yml:68]: This should be `upgrade_d7_node_complete_event_site`

File: config/sync/migrate_plus.migration.upgrade_d7_user_membership.yml [hunk 5]
```diff
@@ -77,4 +77,6 @@ migration_dependencies:
   required:
     - upgrade_d7_user
     - upgrade_d7_node_complete_group
+    - upgrade_d7_node_complete_organisation
+    - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline config/sync/migrate_plus.migration.upgrade_d7_user_membership.yml:81]: This should be `upgrade_d7_node_complete_event_site`

File: lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_declined_invitations.yml [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      method: row
+      message: 'User not found'
+  gid:
+    -
+      plugin: migration_lookup
+      source: gid
+      migration:
+        - upgrade_d7_node_complete_group
+        - upgrade_d7_node_complete_organisation
+        - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_declined_invitations.yml:46]: This should be `upgrade_d7_node_complete_event_site`

File: lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+      method: row
+      message: 'User not found'
+  gid:
+    -
+      plugin: migration_lookup
+      source: gid
+      migration:
+        - upgrade_d7_node_complete_group
+        - upgrade_d7_node_complete_organisation
+        - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml:46]: This should be `upgrade_d7_node_complete_event_site`

File: lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  changed: created
+
+destination:
+  plugin: 'entity:group_content'
+migration_dependencies:
+  required:
+    - upgrade_d7_user
+    - upgrade_d7_node_complete_group
+    - upgrade_d7_node_complete_organisation
+    - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml:76]: This should be `upgrade_d7_node_complete_event_site`

File: lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_pending_visitor_invitations.yml [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+      method: row
+      message: 'User not found'
+  gid:
+    -
+      plugin: migration_lookup
+      source: inv_group_id
+      migration:
+        - upgrade_d7_node_complete_group
+        - upgrade_d7_node_complete_organisation
+        - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_pending_visitor_invitations.yml:42]: This should be `upgrade_d7_node_complete_event_site`

File: lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_pending_visitor_invitations.yml [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+  changed: inv_updated
+
+destination:
+  plugin: 'entity:group_content'
+migration_dependencies:
+  required:
+    - upgrade_d7_user
+    - upgrade_d7_node_complete_group
+    - upgrade_d7_node_complete_organisation
+    - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_group_pending_visitor_invitations.yml:65]: This should be `upgrade_d7_node_complete_event_site`

File: lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_user_membership.yml [hunk 11]
```diff
@@ -75,6 +75,7 @@ destination:
 migration_dependencies:
   required:
     - upgrade_d7_user
-    # TODO: Add the remaining group migration once available.
     - upgrade_d7_node_complete_group
+    - upgrade_d7_node_complete_organisation
+    - upgrade_d7_node_complete_event
```
Comment (4hardfire) [inline lib/modules/eic_migrate/config/install/migrate_plus.migration.upgrade_d7_user_membership.yml:80]: This should be `upgrade_d7_node_complete_event_site`

**Review by 4hardfire — CHANGES_REQUESTED** [review]

**Review by 4hardfire — APPROVED** [review]

File: config/sync/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml (hunk 1 above)
Comment (lramarojaona) [reply config/sync/migrate_plus.migration.upgrade_d7_group_pending_invitations.yml]: I fixed all the wrong migration names

**Review by 4hardfire — APPROVED** [review]
Seems ok to me!

**Pull request merged**